scripts/quests/otherAreas/Knocking_on_Forbidden_Doors.lua

```diff
@@ -0,0 +1,277 @@
+
+quest.sections =
+{
+    -- START: Talk to Enaremand (J-7) on the upper level in Tavnazian Safehold
+    -- QUEST AVAILABLE
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:hasCompletedQuest(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.BEHIND_THE_SMILE)
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Enaremand'] = quest:progressEvent(535),
+
+            onEventFinish =
+            {
+                [535] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    -- QUEST ACCEPTED
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Chemioue'] =
+            {
+                onTrigger = function(player, csid, option, npc)
+                    if quest:getVar(player, 'prog') == 0 then
+                        return quest:event(536)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [536] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 1)
+                end,
+            },
+        },
+
+        [xi.zone.PHOMIUNA_AQUEDUCTS] =
+        {
+            ['Wooden_Ladder'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'prog') == 1 and
+                    npc:getID() == ID.npc.LADDER_KNOCKING then -- Find a better way to identify exactly which ladder it is in case IDs shift
+                        return quest:progressEvent(38, 27) -- Essentially chains the ladder cs with the actual quest cs
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [38] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 2)
+                end,
+            },
+        },
+
+        [xi.zone.MISAREAUX_COAST] =
+        {
+            ['qm_mire_incense'] =
+            {
+                onTrigger = function(player, npc)
+                    --- ??? by end of river
+                    if quest:getVar(player, 'prog') == 2 and
+                    not player:hasKeyItem(xi.ki.MIRE_INCENSE) then
+                        player:addKeyItem(xi.ki.MIRE_INCENSE)
+                        return quest:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.MIRE_INCENSE)
+                    end
+                end,
+            },
+
+            ['qm_alsha'] =
+            {
+                onTrigger = function(player, npc)
+                    -- First time clicking on ???
+                    if player:hasKeyItem(xi.ki.MIRE_INCENSE) and
+                    quest:getVar(player, 'prog') == 2 then
```

Comment:
This var is being pulled multiple times.  Should set a local above the condition